fix(transformer/styled-components): preserve spaces between consecutive interpolations in CSS minification

Dunqing opened the pull request:
Fixes #13312

When minifying CSS in styled-components, spaces between consecutive interpolations were incorrectly removed, breaking CSS selectors.

**Before:** `&:hover ${A}${B}` (incorrect - space removed)
**After:** `&:hover ${A} ${B}` (correct - space preserved)

The fix modifies the whitespace handling logic to preserve spaces between consecutive interpolations while still removing unnecessary spaces elsewhere.

Dunqing commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13346?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#13346** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13346?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13346?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#13327** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13327?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=13346).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

overlookmotel commented on crates/oxc_transformer/src/plugins/styled_components.rs:
```diff
@@ -1062,20 +1062,36 @@ fn minify_template_literal<'a>(lit: &mut TemplateLiteral<'a>, ast: AstBuilder<'a
+                    //   are significant in CSS. ` :hover` (descendant pseudo-selector) is different
+                    //   from `:hover` (direct pseudo-selector). Example: `.parent :hover` selects any
+                    //   hovered descendant, while `.parent:hover` selects the parent when hovered.
+                    && (i == bytes.len() || !matches!(bytes[i], b'{' | b'}' | b',' | b';'))
```
I'm not completely sure, but could we omit this whitespace character if next char is whitespace too?

```suggestion
                    && bytes.get(i).is_none_or(|&b| {
                        !b.is_ascii_whitespace() && !matches!(b, b'{' | b'}' | b',' | b';')
                    })
```

It would make a difference in a case like `something:     {`.

overlookmotel replied:
I'm not sure what patterns this would cover exactly. I'll try it out and do a follow-up PR.

Dunqing committed:
fix(transformer/styled-components): preserve spaces between consecutive interpolations in CSS minification (#13346)

Fixes #13312

When minifying CSS in styled-components, spaces between consecutive interpolations were incorrectly removed, breaking CSS selectors.

**Before:** `&:hover ${A}${B}` (incorrect - space removed)
**After:** `&:hover ${A} ${B}` (correct - space preserved)

The fix modifies the whitespace handling logic to preserve spaces between consecutive interpolations while still removing unnecessary spaces elsewhere.
